### PR TITLE
Update from voxpupuli modulesync_config

### PR DESCRIPTION
**Pull request by nibalizer**

### CODE REVIEW

File: manifests/console.pp [hunk 1]
```diff
@@ -24,6 +24,13 @@
     $template = 'bacula/bconsole.conf.erb'
   ) {
 
+  if $console_package == '' {
+    real_console_package = undef
+  else {
+    real_console_package = Package['bacula-console']
```
Comment (hunner) [inline manifests/console.pp]: These lines need `$` signs?

**Comment by jyaworski** [comment]
Wow, this is rough.

File: spec/classes/bacula_spec.rb [hunk 2]
```diff
@@ -16,6 +16,44 @@
...
+        :bacula_manage_db_tables        => '',
+        :bacula_db_backend              => 'mysql',
+        :bacula_mail_to                 => '',
+        :bacula_director_password       => '',
+        :bacula_console_password        => '',
+        :bacula_director_server         => '',
+        :bacula_storage_server          => '',
+        :bacula_director_package        => 'bacula-director',
+        :bacula_storage_package         => 'bacula-storage',
+        :bacula_client_package          => 'bacula-client',
```
Comment (hunner) [inline spec/classes/bacula_spec.rb]: This is being picked up by `generate_clients()` call of `to_hash` even though it's not related.

File: lib/puppet/parser/functions/generate_clients.rb [hunk 3]
```diff
@@ -5,27 +5,26 @@ module Puppet::Parser::Functions
...
+            raise Puppet::ParseError, 'Could not parse parameters given. Please check your format'
+          end
+          if [nil, ''].include?(val_array[0]) || [nil, ''].include?(val_array[1])
+            raise Puppet::ParseError, 'Could not parse parameters given. Please check your format'
+          end
+          val_array
+        end]
+      rescue
+        raise Puppet::ParseError, 'Could not parse parameters given. Please check your format'
       end
```
Comment (hunner) [inline lib/puppet/parser/functions/generate_clients.rb]: All of the above code is terrible and should be deleted.
This function takes an argument as `args` but that in handled in the second block below... What is this block doing then? It calls `to_hash` seemingly out of no where, and that takes the catalog's scope and turns it into a hash, supposedly to look for variables "at top scope" that come from some data source (like the old puppet enterprise console) but it has no safety and will pick up ANY variables in the namespace, including facts. This is a terrible design and seriously bad side-effects.
I mean, this function says "okay, I was called with `args` but before using that I'm just going to see if I can find anything else that could possibly have been passed to me but wasn't" and then tries to execute it.
Also, the raises in the begin just get captured by the rescue block that then raises again with the same message. Wat.

**Comment by dhoppe** [comment]
@hunner According to the readme, we just need to populate the class parameter `clients` at the spec files. But the readme is inconclusive about the format (class parameter vs dashboard), because the function will skip anything unless it is beginning with `bacula_client_`. 

In the end it is just passing parameters to a define which creates a configuration for each client, based on  the fileset, schedule and template.

**Comment by bastelfreak** [comment]
I'm going to merge this as is. The module needs a huge refactoring but that is out of scope for a modulesync.

**Pull request merged**